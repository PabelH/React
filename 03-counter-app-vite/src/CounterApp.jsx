import PropTypes  from "prop-types"

const CounterApp = ({value}) => {
  return (
    <>
    <h1>CounterApp</h1>
    <h2> { value } </h2>

    <button onClick={ function() {console.log('Hello Pabello')} }>
      +1
    </button>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
    
  }
  

export default CounterApp
