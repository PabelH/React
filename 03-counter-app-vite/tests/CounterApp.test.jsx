import CounterApp from "../src/CounterApp";
import { render, screen } from "@testing-library/react";


describe('Test in <CounterApp />', () => {

    const initialValue = 100;

    test('this test must match the snapshot"',()=>{

        const {container} = render(<CounterApp value={ initialValue }/>);
        expect( container ).toMatchSnapshot();

    });
    test('This test should show the initial value of 100"',()=>{

        const {container} = render(<CounterApp value={ initialValue }/>);
        expect( container ).toMatchSnapshot();

    });
});