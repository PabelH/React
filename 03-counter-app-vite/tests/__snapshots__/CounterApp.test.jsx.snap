// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <CounterApp /> This test should show the initial value of 100" 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button>
     Reset 
  </button>
</div>
`;

exports[`Test in <CounterApp /> this test must match the snapshot" 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button>
     Reset 
  </button>
</div>
`;
